brian2/tests/test_synapses.py

```diff
@@ -2190,6 +2191,22 @@ def test_synapses_refractory_rand():
     assert all(target.v[:5] > 0)
     assert_equal(target.v[5:], 0)
 
+
+@attr('codegen-independent')
+def test_range_noint_error():
+    # arguments to `range` should only be integers (issue #781)
+    G = NeuronGroup(42, 'v: 1', threshold='False')
+    S = Synapses(G, G, 'w:1', 'v+=w')
+    msg = 'The "{}" argument of the range function was .+, but it needs to be an integer.'
+    assert_raises_regex(TypeError, msg.format('high'), lambda: S.connect(j='k for k in range(42.0)'))
```

Comment:
Good test, I wonder if it is not a bit too specific by matching the exact text of the error message. On the other hand, it should be pretty obvious what the issue is, if we ever to decide to change the text. So, ok, let's leave it that way.